src/sniffer/InterfaceHandler.c

```diff
@@ -0,0 +1,114 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <pcap.h>
+
+#include "InterfaceHandler.h"
+#include "ErrorHandler.h"
+#include "Sniffer.h"
+
+pcap_if_t *InterfaceHandler_availableInterfaces;
+pcap_if_t *InterfaceHandler_currentInterface;
+pcap_t *InterfaceHandler_captureHandler;
+
+void InterfaceHandler_Init() {
+    // Find list of interfaces available in the system
+    if(pcap_findalldevs(&InterfaceHandler_availableInterfaces, ErrorHandler_GetPcapErrorBuffer()) == PCAP_ERROR) {
+        ErrorHandler_DisplayPcapErrorAndExit("Error scanning for devices", true);
+    }
+}
+
+InterfaceHandler_status_t InterfaceHandler_SelectInterface(int interfaceNumber) {
+    int index;
+    pcap_if_t *currentIf;
+
+    if (InterfaceHandler_availableInterfaces == NULL) {
+        ErrorHandler_DisplayErrorAndExit("[InterfaceHandler] Interfaces not available!");
+    } else {
+        for (currentIf = InterfaceHandler_availableInterfaces, index = 0; currentIf != NULL; currentIf = currentIf->next, index++) {
+            if (interfaceNumber == index) {
+                InterfaceHandler_currentInterface = currentIf;
+                return InterfaceHandler_status_OK;
+            }
+        }
+    }
+
+    return InterfaceHandler_status_ERROR;
+}
+
+InterfaceHandler_status_t InterfaceHandler_OpenCapture() {
+    if (InterfaceHandler_currentInterface == NULL) {
+        ErrorHandler_DisplayWarning("[InterfaceHandler] Current interface not available!");
+        return InterfaceHandler_status_ERROR;
+    } else {
+        // Prepare capture
+        if ((InterfaceHandler_captureHandler = pcap_open_live(InterfaceHandler_currentInterface->name,	// Name of the device.
+                                BUFSIZ,			                                                        // Portion of the packet to capture. 
+                                                                                                        // Value 65536 grants that the whole packet is captured on all the MACs.
```

Comment:
missing one argument here?

Reply:
Both comments of ln 46 and 47 are for BUFSIZE